lib/ansible/parsing/splitter.py

```diff
@@ -188,6 +170,17 @@ def split_args(args):
     block_depth = 0  # used to count nested jinja2 {% %} blocks
     comment_depth = 0  # used to count nested jinja2 {# #} blocks
 
+    # the list of params parsed out of the arg string
+    # this is going to be the result value when we are done
+    params = []
+
+    # Initial split on newlines
+    items = args.split('\n')
+
+    # If we start with a newline, ensure we have it preserved
+    if args.startswith('\n'):
+        params.append('')
```

Comment:
I'd move this up just below `params = []` to keep it with the initialization of params in it's original location.

Reply:
Well, the reason it is here it's because it relates to the splitting by newline.
That's why I preferred this order.